### PR TITLE
[Inbox] reload data on re-entering tab

### PR DESCRIPTION
**Pull request by l2succes**
Fixes: https://github.com/artsy/collector-experience/issues/794

Depends on: https://github.com/artsy/eigen/pull/2479

### CODE REVIEW

**Comment by ArtsyOpenSource** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Fails</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:no_entry_sign:</td>
      <td>

  No CHANGELOG added.
  </td>
    </tr>
  </tbody>
</table>




<p align="right">
  Generated by :no_entry_sign: <a href="http://github.com/danger/danger-js/">dangerJS</a>
</p>

**Comment by ArtsyOpenSource** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Fails</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:no_entry_sign:</td>
      <td>

  No CHANGELOG added.
  </td>
    </tr>
  </tbody>
</table>




<p align="right">
  Generated by :no_entry_sign: <a href="http://github.com/danger/danger-js/">dangerJS</a>
</p>

File: src/lib/Containers/Inbox.tsx [hunk 1]
```diff
@@ -31,6 +32,12 @@ export class Inbox extends React.Component<Props, State> {
     this.fetchData = this.fetchData.bind(this)
   }
 
+  componentWillReceiveProps(newProps) {
```
Comment (mennenia) [inline src/lib/Containers/Inbox.tsx]: So just checking I understand it right: 

Whenever you navigate to the messaging inbox, the property `isVisible` is set in the props, which means the props are updated, and `componentWillReceiveProps` is triggered, upon which any new data can be fetched? And then _just_ the new messages will be fetched and rendered/added as opposed to the whole VC loaded fresh every time (which we do now and is what is generally being fixed here).

**Comment by mennenia** [comment]
Ahhh 😍😍😍  so coooool

File: src/lib/Containers/Inbox.tsx (hunk 1 above)
Comment (l2succes) [reply src/lib/Containers/Inbox.tsx]: Yes, that's precisely what happens 😊.

Comment (mennenia) [reply src/lib/Containers/Inbox.tsx]: 🎉

**Pull request merged**